### ISSUE TITLE
Support more LMS

### ISSUE THREAD

**Issue by jtcheng26** [jtcheng26/canvas-task-extension#55]
- D2L
- Blackboard
- Google Classroom
- Schoology
- Moodle

Canvas is most popular in higher education, but only around 25-50% of the total market share.
Should these be separate extensions or part of this one?

**Comment by ed789d0** [comment]
I think the code is different enough for each LMS that they should be separate extensions.

**Comment by jtcheng26** [comment]
That's true, although it would just be the api requests and subsequent data wrangling that would need to be modified. Should check and see if the other LMS have public API documentation.

**Comment by ed789d0** [comment]
For schoology: https://developers.schoology.com/api

**Comment by ausward** [comment]
Moodle has APIs but it looks like they are more for schools/university sys-admins 
https://moodledev.io/docs/apis

**Comment by jtcheng26** [comment]
Update: I want to focus on this more in the next few months, ideally starting with Gradescope integration then Blackboard. Gradescope because I have access to it already so it would be a nice test to see how much needs to change in the code, Blackboard because it's the most requested.

**Comment by ed789d0** [comment]
I don't know if Gradescope has an API but in [my experience](https://github.com/ed789d0/enhance-gradescope/blob/main/content_script.js) it's really easy to scrape.
